### DISCUSSION TITLE
Vertical Scrolling for Comics

### DISCUSSION THREAD

**Discussion by gemdo** [jellyfin/jellyfin-web#5606] [Ideas]
Some Comics (.cbz in my case) do not display properly in the current reader, as they are designed for each chapter to be read as a long, vertically scrolling, page similar to the Webtoon layout. 

This would be similar to and expand the work of [PR #3898](https://github.com/jellyfin/jellyfin-web/pull/3898) from a few years ago modifying page layout/changes, and would implement a comic view similar to the vertical layout in the [JellyBook project](https://github.com/JellyBookOrg/JellyBook).

I would be happy to work on this myself, but I just wanted to check in first to see if anyone is aware of current work on a similar feature (I wasn't able to find anything), or if anyone has feedback that would be good to consider in development.

Thanks!

**Comment by OmegaGiven** [comment]
You can apply this custom css to your server via dashboard>Branding  and it will allow scroll on both desktop and mobile but the issue with mobile is it will take the initial tap as a change page then you scroll so not a great solution for mobile but works for desktop:

/* Ensure the book viewer only affects the Book Library section */
body.libraryDocument #pdfPlayer {
    max-height: 90vh; /* 90% of the viewport height */
    overflow-y: auto; /* Enables vertical scrolling */
    padding: 0; /* Remove extra padding to allow full use of vertical space */
    display: flex;
    flex-direction: column;
    touch-action: pan-y; /* Allows vertical scrolling (pan-y) */
}

/* Ensure the canvas inside the PDF viewer scales properly */
body.libraryDocument #pdfPlayer canvas {
    width: 100%; /* Ensure the canvas fits the container width */
    height: 100%; /* Allow the canvas to take up full height */
    object-fit: contain; /* Ensures the aspect ratio is maintained */
}

/* Optional: Style for the close button */
body.libraryDocument #pdfPlayer .btnExit {
    background-color: transparent; /* Transparent background for button */
    color: #333; /* Button color */
}

**Reply by Jstudner** [reply]
Did anyone ever get this working? I tried your custom css but dont see a difference.

**Reply by OmegaGiven** [reply]
Yea custom CSS is finicy since its a Browser cookie thing so you may need to delete cookies and such and the custom CSS may work after that but i do recognize it may not work every time. really our best option is to use other apps for books until these are fixed. im waiting for my PR to fix pdfs for the past few months as pdfs have no functionality other than display with no zoom or anything.

**Reply by Jstudner** [reply]
yea, I am getting close to making a plugin. never done much jellyfin code, didnt want to mess with it more than I needed to. But not much out there that handles webtoon format... and even if there was I would very much like it integrated!

Thx for the quick responce!
